pkg/common/common.go

```diff
@@ -56,28 +60,29 @@ func FetchEnv(key string, def string) string {
 	return val
 }
 
+// GetDefaultIfNil returns the value of a pointer argument, or a default value if the pointer is nil.
 func GetDefaultIfNil[T any](val *T, def T) T {
 	if reflect.ValueOf(val).IsNil() {
 		return def
 	}
 	return *val
 }
 
+// GetEmptySliceIfNil returns a provided slice, or an empty slice of the same type if the input slice is nil.
 func GetEmptySliceIfNil[T any](val []T) []T {
-	if val == nil {
-		return make([]T, 0)
+	if len(val) == 0 {
+		return []T{}
 	}
 	return val
 }
 
 // NamespacedNameToObjectKey converts <namespace/name> format string to k8s object key.
 // It's common for K8s to reference an object using this format. For e.g. gateways in VirtualService.
 func NamespacedNameToObjectKey(namespacedName, defaultNamespace string) client.ObjectKey {
-	split := strings.Split(namespacedName, "/")
-	if len(split) == 2 {
-		return client.ObjectKey{Name: split[1], Namespace: split[0]}
+	if i := strings.IndexRune(namespacedName, '/'); i >= 0 {
```

Comment:
Is there any difference here comparing to the "old" implementation?

Reply:
Yes, indeed it is! According to results of the benchmark I did, this new implementation is approximately 12-13 times faster than the previous implementation.

strings.IndexRune() is a more lightweight operation than strings.Split(). When you call strings.Split() on a string, the Go runtime must do a significant amount of work to split the string into an array. In contrast, when you call strings.IndexRune(), the runtime only needs to find the index of the first occurrence of a single character in the string.

Hope, that answeres your question, @eguzki! :angel:

Reply:
As I mentioned before, always better to not include changes when writing tests... however, since this benchmark is considerable faster than the other, I'd say to leave it...

Reply:
Next time I'll create a separate pull request! 🫡🫡🫡